File: examples/python/stateful-wots-gate/StatefulWOTSGate.runar.py
from runar import StatefulSmartContract, ByteString, Bigint, public, assert_, verify_wots


# StatefulWOTSGate — stateful + post-quantum interaction fixture (GAP-407).
class StatefulWOTSGate(StatefulSmartContract):
    count: Bigint

    def __init__(self, count: Bigint):
        super().__init__(count)
        self.count = count

    @public
    def advance(self, msg: ByteString, sig: ByteString, wots_pub_key: ByteString):
        assert_(verify_wots(msg, sig, wots_pub_key))
        self.count = self.count + 1

File: examples/ruby/stateful-wots-gate/StatefulWOTSGate.runar.rb
require 'runar'

# StatefulWOTSGate — stateful + post-quantum interaction fixture (GAP-407).
class StatefulWOTSGate < Runar::StatefulSmartContract
  prop :count, Bigint

  def initialize(count)
    super(count)
    @count = count
  end

  runar_public msg: ByteString, sig: ByteString, wots_pub_key: ByteString
  def advance(msg, sig, wots_pub_key)
    assert verify_wots(msg, sig, wots_pub_key)
    @count = @count + 1
  end
end
